adapters/tlsio_mbedtls.c

```diff
@@ -343,52 +348,86 @@ static int on_io_send(void *context, const unsigned char *buf, size_t sz)
     return result;
 }
 
-static int tlsio_entropy_poll(void* v, unsigned char* output, size_t len, size_t* olen)
+static int tlsio_entropy_poll(void *v, unsigned char *output, size_t len, size_t *olen)
 {
     (void)v;
     int result = 0;
     srand((unsigned int)time(NULL));
-    for (uint16_t i = 0; i < len; i++) {
+    for (uint16_t i = 0; i < len; i++)
+    {
         output[i] = rand() % 256;
     }
     *olen = len;
     return result;
 }
 
-static void mbedtls_init(TLS_IO_INSTANCE* instance, const char *host)
+// Un-initialize mbedTLS
+static void mbedtls_uninit(TLS_IO_INSTANCE *tls_io_instance)
 {
+    if (tls_io_instance->tls_status > 0)
+    {
+        // mbedTLS cleanup...
+        mbedtls_ssl_free(&tls_io_instance->ssl);
+        mbedtls_ssl_config_free(&tls_io_instance->config);
+        mbedtls_x509_crt_free(&tls_io_instance->trusted_certificates_parsed);
+        mbedtls_ctr_drbg_free(&tls_io_instance->ctr_drbg);
+        mbedtls_entropy_free(&tls_io_instance->entropy);
+
+        tls_io_instance->tls_status = 0;
+    }
+}
+
+// Initialize mbedTLS
+static void mbedtls_init(TLS_IO_INSTANCE *tls_io_instance)
+{
+    if (tls_io_instance->tls_status == 1)
```

Comment:
>1 [](start = 39, length = 1)

magic numbers for tls status. please replace with appropriate enums

Reply:
fixed